### PR TITLE
fix: score rank & ranked score modification

### PR DESCRIPTION
**Pull request by tsunyoku**

### CODE REVIEW

File: app/repositories/leaderboards.py [hunk 1]
```diff
@@ -185,8 +184,6 @@ async def fetch_user_score(
                 row_number() OVER (PARTITION BY s.userid ORDER BY s.{sort_column} DESC) score_order_rank
             FROM {scores_table} s
             INNER JOIN users ON users.id = s.userid
-            LEFT JOIN clans c ON users.clan_id = c.id
-            INNER JOIN users_stats ON users_stats.id = s.userid
```
Comment (cmyui) [inline app/repositories/leaderboards.py:189]: these are in other queries too - should they also be removed?

File: app/usecases/leaderboards.py [hunk 2]
```diff
@@ -78,22 +78,21 @@ async def fetch_beatmap_leaderboard(
...
-
-    for idx, leaderboard_score in enumerate(leaderboard_scores):
-        if score_to_judge.mode > Mode.MANIA:
-            to_beat = leaderboard_score["pp"]
-        else:
-            to_beat = leaderboard_score["score"]
-
-        if comparison_value > to_beat:
-            return idx + 1
+    print(score_id_to_judge)
```
Comment (cmyui) [inline app/usecases/leaderboards.py]:
```suggestion
```

**Comment by cmyui** [comment]
tested locally on a couple of maps?

File: app/repositories/leaderboards.py (hunk 1 above)
Comment (tsunyoku) [reply app/repositories/leaderboards.py:189]: removed where possible

**Comment by tsunyoku** [comment]
> tested locally on a couple of maps?

yes

**Comment by cmyui** [comment]
are we still showing clan names on lbs? just remembered that removing it from the join feels wrong - i remember this being needed

is it replaced by some other code performing the same job?

**Review by cmyui — APPROVED** [review]

**Comment by tsunyoku** [comment]
> are we still showing clan names on lbs? just remembered that removing it from the join feels wrong - i remember this being needed
> 
> 
> 
> is it replaced by some other code performing the same job?

we are, but not for the user's personal best. the join is still there for fetching leaderboard scores, but not the user's personal best

**Pull request merged**